### PR TITLE
API ingestion keys - CRUD

### PR DESCRIPTION
**Pull request by YounixM**
image
image
image
image
image

## Summary by CodeRabbit

- **New Features**
    - Added functionality to create, retrieve, update, and delete API keys.
    - Implemented a search feature for easily finding specific API keys.
    - Enhanced the UI with new styles for managing API keys, including modals and detailed key information display.

### CODE REVIEW

File: frontend/src/container/APIKeys/APIKeys.tsx [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+		error,
+		isRefetching,
+		refetch: refetchAPIKeys,
+	} = useGetAllAPIKeys();
+
+	useEffect(() => {
+		setDataSource(APIKeys?.data.data || []);
+	}, [APIKeys?.data.data]);
+
+	useErrorNotification(error);
```
Comment (Vikrant2520) [inline frontend/src/container/APIKeys/APIKeys.tsx:132]: can add if condition to call the hook only when error is present.

File: frontend/src/container/APIKeys/APIKeys.tsx [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+			>
+				<Typography.Text className="delete-text">
+					{t('delete_confirm_message', {
+						keyName: activeAPIKey?.name,
+					})}
+				</Typography.Text>
+			</Modal>
+
+			{/* Edit Key Modal */}
+			<Modal
```
Comment (Vikrant2520) [inline frontend/src/container/APIKeys/APIKeys.tsx:428]: can extract the edit and add new code to a component and render the edit / add conditonally based on param

File: frontend/src/container/APIKeys/APIKeys.tsx [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+							name="expiration"
+							label="Expiration"
+							rules={[{ required: true }]}
+						>
+							<Select
+								className="expiration-selector"
+								placeholder="Expiration"
+								options={[
+									{
+										value: '1',
```
Comment (Vikrant2520) [inline frontend/src/container/APIKeys/APIKeys.tsx:627]: can move this to enum

File: frontend/src/container/APIKeys/APIKeys.styles.scss [hunk 4]
```diff
@@ -0,0 +1,510 @@
+.api-key-container {
+	margin-top: 32px;
```
Comment (Vikrant2520) [inline frontend/src/container/APIKeys/APIKeys.styles.scss:2]: we might need to handle light mode here

**Comment by Vikrant2520** [comment]
image
shall we show here what kind of roles the key  holds ?

**Comment by Vikrant2520** [comment]
image
Key and ID seems to be duplicate and the copy on ID copies a number rather than the shown text

**Comment by YounixM** [comment]
@Vikrant2520: API Response data is not as expected. Waiting for that to be fixed and deployed. Will update data related review comments post that.

File: frontend/src/container/APIKeys/APIKeys.tsx (hunk 2 above)
Comment (YounixM) [reply frontend/src/container/APIKeys/APIKeys.tsx:428]: The modal triggers happen via the listing page, so controlling the modals from the nested components will be back and forth between the child and parent. Hence this way.

File: frontend/src/container/APIKeys/APIKeys.styles.scss [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+	border-radius: 20px;
+	padding: 0px 12px;
+	background: var(--bg-ink-200, #23262e);
+
+	.copy-key-btn {
+		cursor: pointer;
+	}
+}
+
+.lightMode {
```
Comment (YounixM) [inline frontend/src/container/APIKeys/APIKeys.styles.scss:413]: @Vikrant2520: light mode is handled here.  Will verify in lightMode and darkMode.

File: frontend/src/container/APIKeys/APIKeys.tsx (hunk 1 above)
Comment (YounixM) [reply frontend/src/container/APIKeys/APIKeys.tsx:132]: You mean creation would happen only when the error occurs?

File: frontend/src/container/APIKeys/APIKeys.tsx (hunk 3 above)
Comment (YounixM) [reply frontend/src/container/APIKeys/APIKeys.tsx:627]: Yes, the payload for handling expiry is dependent on the API side. Will update this once the datatype is finalized.

File: frontend/src/api/APIKeys/createAPIKey.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import axios from 'api';
```
Comment (YounixM) [inline frontend/src/api/APIKeys/createAPIKey.ts:1]: @coderabbitai generate unit tests for this file.

File: frontend/src/container/APIKeys/APIKeys.tsx [hunk 7]
```diff
@@ -0,0 +1,747 @@
+import './APIKeys.styles.scss';
```
Comment (YounixM) [inline frontend/src/container/APIKeys/APIKeys.tsx:1]: @coderabbitai generate unit tests for this file using jest, msw.

Comment (YounixM) [reply frontend/src/container/APIKeys/APIKeys.tsx:1]: @coderabbitai: Generate unit test cases for CRUD operations for the API Keys.

File: frontend/src/container/APIKeys/APIKeys.tsx [hunk 8]
```diff
@@ -0,0 +1,831 @@
...
+
+	useEffect(() => {
+		setActiveAPIKey(APIKeys?.data.data[0]);
+	}, [APIKeys]);
+
+	useEffect(() => {
+		setDataSource(APIKeys?.data.data || []);
+	}, [APIKeys?.data.data]);
+
+	useEffect(() => {
```
Comment (Rajat-Dabade) [inline frontend/src/container/APIKeys/APIKeys.tsx:165]: We have hook for this `useErrorNotification`, can be used here.

Comment (YounixM) [reply frontend/src/container/APIKeys/APIKeys.tsx:165]: We are using useErrorNotification itself here @Rajat-Dabade

**Review by Vikrant2520 — APPROVED** [review]

**Review by Vikrant2520 — APPROVED** [review]

**Review by Rajat-Dabade — APPROVED** [review]
LGTM 🚀

**Pull request merged**